packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -748,88 +713,38 @@ abstract class BaseFlutterTask extends DefaultTask {
 
         intermediateDir.mkdirs()
 
-        if (buildMode == "profile" || buildMode == "release") {
-            project.exec {
-                executable flutterExecutable.absolutePath
-                workingDir sourceDir
-                if (localEngine != null) {
-                    args "--local-engine", localEngine
-                    args "--local-engine-src-path", localEngineSrcPath
-                }
-                args "build", "aot"
-                args "--suppress-analytics"
-                args "--quiet"
-                args "--target", targetPath
-                args "--output-dir", "${intermediateDir}"
-                args "--target-platform", "${targetPlatform}"
-                if (extraFrontEndOptions != null) {
-                    args "--extra-front-end-options", "${extraFrontEndOptions}"
-                }
-                if (extraGenSnapshotOptions != null) {
-                    args "--extra-gen-snapshot-options", "${extraGenSnapshotOptions}"
-                }
-                args "--${buildMode}"
-            }
-        }
-
         project.exec {
             executable flutterExecutable.absolutePath
             workingDir sourceDir
-
             if (localEngine != null) {
                 args "--local-engine", localEngine
                 args "--local-engine-src-path", localEngineSrcPath
             }
-            args "build", "bundle"
-            args "--target", targetPath
-            args "--target-platform", "${targetPlatform}"
             if (verbose) {
                 args "--verbose"
+            } else {
+                args "--quiet"
             }
-            if (fileSystemRoots != null) {
-                for (root in fileSystemRoots) {
-                    args "--filesystem-root", root
-                }
-            }
-            if (fileSystemScheme != null) {
-                args "--filesystem-scheme", fileSystemScheme
-            }
-            if (trackWidgetCreation) {
-                args "--track-widget-creation"
-            }
-            if (compilationTraceFilePath != null) {
-                args "--compilation-trace-file", compilationTraceFilePath
-            }
-            if (createPatch) {
-                args "--patch"
-                args "--build-number", project.android.defaultConfig.versionCode
-                if (buildNumber != null) {
-                    assert buildNumber == project.android.defaultConfig.versionCode
-                }
-            }
-            if (baselineDir != null) {
-                args "--baseline-dir", baselineDir
-            }
+            args "assemble"
+            args "--depfile", "${intermediateDir}/flutter_build.d"
+            args "--output", "${intermediateDir}"
+            args "-dTargetFile=${targetPath}"
```

Comment:
FWIW: single dash flags are used for abbreviations. Also, most of the tool's flags use hyphen-separated flags.

Reply:
These aren't flags though, it's a multi option called `--define` with a value of `KeyName=Value`. This allows assemble to take arbitrary key/value pairs without needing to be aware/maintain a large set of arguments. The fact that the key names are pascal-case distinguishes them from command line arguments.

Reply:
alright. sounds good.